routes/web/project/github.rb

```diff
@@ -39,5 +39,15 @@ class CloverWeb
         r.redirect "https://github.com/apps/#{Config.github_app_name}/installations/new", 302
       end
     end
+
+    r.on "cache" do
+      r.get true do
+        repository_id_q = @project.github_installations_dataset.join(:github_repository, installation_id: :id).select(Sequel[:github_repository][:id])
+        @entries = Serializers::GithubCacheEntry.serialize(GithubCacheEntry.where(repository_id: repository_id_q).exclude(committed_at: nil).eager(:repository).order(Sequel.desc(:created_at)).all)
```

Comment:
radical, a dataset composition (passing one into another) to get a subquery? do I have that right?

Reply:
Yeah, it returns `Sequel::Dataset` instead of an array when you use `select` instead of `select_map`, allowing you to use it as a subquery in where filtering.

Reply:
```ruby
[19] clover-development(main)> repository_id_q = @project.github_installations_dataset.join(:github_repository, installation_id: :id).select(Sequel[:github_repository][:id])
=> #<Sequel::Postgres::Dataset: "SELECT \"github_repository\".\"id\" FROM \"github_installation\" INNER JOIN \"github_repository\" ON (\"github_repository\".\"installation_id\" = \"github_installation\".\"id\") WHERE (\"github_installation\".\"project_id\" = $1); [\"ed0b26bf-53c4-82d2-9a00-21e5f05dc364\"]">

[20] clover-development(main)> GithubCacheEntry.where(repository_id: repository_id_q).exclude(committed_at: nil).eager(:repository).order(Sequel.desc(:created_at)).sql

SELECT * 
FROM "github_cache_entry" 
WHERE (
  (
    "repository_id" IN (
      SELECT "github_repository"."id" 
      FROM "github_installation" 
      INNER JOIN "github_repository" ON ("github_repository"."installation_id" = "github_installation"."id") 
      WHERE ("github_installation"."project_id" = $1)
    )
  ) 
  AND ("committed_at" IS NOT NULL)
) 
ORDER BY "created_at" DESC


```

Reply:
I learned this from @furkansahin with this line https://github.com/ubicloud/ubicloud/blob/3ed6a2fa4f0a4e7a389cadcdf3b66fd78584cac9/model/vm_pool.rb#L19